File: 1752-check-if-array-is-sorted-and-rotated/main.go
func check(nums []int) bool {
    n := len(nums)

    sorted_nums := make([]int, n)
    copy(sorted_nums, nums)
    sort.Ints(sorted_nums)

    for i := range nums {
        isMatch := true
        
        for j := range nums {
            if sorted_nums[j] != nums[(i + j) % n] {
                isMatch = false
                break
            }
        }

        if isMatch {
            return true
        }
    }

    return false
}

File: 1752-check-if-array-is-sorted-and-rotated/main.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        
        size_t n = nums.size();
        vector<int> sorted_nums(nums.begin(), nums.end());
        sort(sorted_nums.begin(), sorted_nums.end());

        for (size_t i = 0; i < n; i++){
            bool is_match = true;

            for (size_t j = 0; j < n; j++) {
                if (sorted_nums[j] != nums[(i + j) % n ]) {
                    is_match = false;
                    break;
                }
            }

            if (is_match) {
                return true;
            }
        }

        return false;
    }
};
